File: backend/schemas/order.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class OrderItemResponse(BaseModel):
    id: UUID
    product_id: UUID
    quantity: int
    price_at_purchase: float
    product_name: str = ""
    product_image: str = ""

    class Config:
        from_attributes = True


class OrderResponse(BaseModel):
    id: UUID
    total_amount: float
    subtotal: float
    shipping: float
    tax: float
    status: str
    stripe_payment_id: str | None = None
    shipping_address: dict | None = None
    items: list[OrderItemResponse] = []
    created_at: datetime

    class Config:
        from_attributes = True


class OrderListResponse(BaseModel):
    orders: list[OrderResponse]
    total: int
    page: int
    pages: int


class OrderStatusUpdate(BaseModel):
    status: str

File: frontend/src/types/order.ts
export enum OrderStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  SHIPPED = 'shipped',
  DELIVERED = 'delivered',
  CANCELLED = 'cancelled',
}

export interface OrderItem {
  id: string
  product_id: string
  quantity: number
  price_at_purchase: number
  product_name: string
  product_image: string
}

export interface Order {
  id: string
  total_amount: number
  subtotal: number
  shipping: number
  tax: number
  status: OrderStatus
  stripe_payment_id?: string
  shipping_address?: Record<string, string>
  items: OrderItem[]
  created_at: string
}

export interface OrderListResponse {
  orders: Order[]
  total: number
  page: number
  pages: number
}
